shlr/gdb/src/gdbserver/core.c

```diff
@@ -49,6 +49,35 @@ static int _server_handle_qOffsets(libgdbr_t *g, gdbr_server_cmd_cb cmd_cb, void
 	return send_msg (g, buf);
 }
 
+static int _server_handle_M(libgdbr_t *g, gdbr_server_cmd_cb cmd_cb, void *core_ptr) {
+	if (send_ack (g) < 0) {
+		return -1;
+	}
+	g->data[g->data_len] = '\0';
+	char *memstr, *buf;
+	ut64 memlen, addr;
+	if (sscanf (g->data + 1, "%"PFMT64x",%"PFMT64x, &addr, &memlen) != 2) {
+		return send_msg (g, "E01");
+	}
+	if (!(memstr = strchr (g->data, ':')) || !*(++memstr)) {
+		return send_msg (g, "E01");
+	}
+	if (memlen * 2 != strlen (memstr)) {
+		return send_msg (g, "E01");
+	}
+	if (!(buf = malloc (memlen * 2 + 64))) {
```

Comment:
Use a variable to hold this memlen*2... and use it jn this malloc and in the snprintf